src/main/java/org/opensearch/searchrelevance/judgments/LlmJudgmentsProcessor.java

```diff
@@ -388,98 +471,127 @@ private void processWithLLM(
 
     private void generateLLMJudgmentForQueryText(
         String modelId,
-        String queryTextWithReference,
+        String queryTextWithCustomInput,
         int tokenLimit,
         List<String> contextFields,
         Map<String, String> unprocessedUnionHits,
         Map<String, String> docIdToRating,
+        String promptTemplate,
+        LLMJudgmentRatingType ratingType,
+        String promptTemplateCode,
         ActionListener<Map<String, String>> listener
     ) {
         log.debug("calculating LLM evaluation with modelId: {} and unprocessed unionHits: {}", modelId, unprocessedUnionHits);
         log.debug("processed docIdToRating before llm evaluation: {}", docIdToRating);
 
         if (unprocessedUnionHits.isEmpty()) {
-            log.info("All hits found in cache, returning cached results for query: {}", queryTextWithReference);
+            log.info("All hits found in cache, returning cached results for query: {}", queryTextWithCustomInput);
             listener.onResponse(docIdToRating);
             return;
         }
 
-        String[] queryTextRefArr = queryTextWithReference.split(DELIMITER);
+        String[] queryTextRefArr = queryTextWithCustomInput.split(DELIMITER, 2);
```

Comment:
why you choose 2 as a threshold for regex, can you put explanation as java code comment